File: bindings/kotlin/examples/SystemState.kt
// Copyright (c) 2025 IOTA Stiftung
// SPDX-License-Identifier: Apache-2.0

import iota_sdk.GraphQlClient
import kotlinx.coroutines.runBlocking

fun main() = runBlocking {
    try {
        val client = GraphQlClient.newDevnet()
        val systemState = client.latestSystemState()
        println("Latest system state: ${systemState.valueAsJson}")
    } catch (e: Exception) {
        e.printStackTrace()
        kotlin.system.exitProcess(1)
    }
}

File: bindings/python/examples/system_state.py
# Copyright (c) 2025 IOTA Stiftung
# SPDX-License-Identifier: Apache-2.0

from lib.iota_sdk_ffi import *

import asyncio


async def main():
    client = GraphQlClient.new_devnet()

    system_state = await client.latest_system_state()
    print("Latest system state:", system_state.value_as_json)


if __name__ == "__main__":
    asyncio.run(main())